modules/distribution/src/repository/resources/conf/templates/repository/conf/tomcat/catalina-server.xml.j2

```diff
@@ -109,6 +109,11 @@
                 <Valve className="org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve" threshold="600"/>
                 <Valve className="org.wso2.carbon.tomcat.ext.valves.CompositeValve"/>
 
+                <!-- OAuthAppTenantResolverValve is required to be invoked when tenant qualified urls are disabled. This should be invoked before the AuthenticationValve -->
+                {% if tenant_context.enable_tenant_qualified_urls is sameas false %}
+                <Valve className="org.wso2.carbon.identity.auth.valve.OAuthAppTenantResolverValve"/>
```

Comment:
don't we have a config to disable the uniqueness of the client ID. if so, this should be enabled based on both the conditions, right?

Reply:
There's no such config. By default client ID tenant uniqueness will be shipped and in both of these cases, underline code expects tenant id to be present when retrieving the client application.